src/System.Private.CoreLib/shared/System/Globalization/TimeSpanFormat.cs

```diff
@@ -155,28 +291,21 @@ private static StringBuilder FormatStandard(TimeSpan value, bool isInvariant, Re
             AppendNonNegativeInt32(sb, minutes, literal.mm);    // mm
             sb.Append(literal.MinuteSecondSep);                 // :
             AppendNonNegativeInt32(sb, seconds, literal.ss);    // ss
-            if (!isInvariant && pattern == Pattern.Minimum)
+            if (!full)
             {
                 int effectiveDigits = literal.ff;
-                while (effectiveDigits > 0)
+                while (effectiveDigits > 0 && fraction % 10 == 0)
```

Comment:
We use mod here and division within the loop. Would using DivRem be faster?

Reply:
> Would using DivRem be faster?

It would require measuring.  I didn't do any substantive changes to this G-formatting code, though, only minor cleanup.